docs/design/architecture.md

```diff
@@ -0,0 +1,29 @@
+# Architecture 
+Kubernetes Console project consists of two main components. They are called here the
+frontend and the backend.
+
+The frontend is a single page web application that runs in a browser. It fetches all its
+business data from the backend using standard HTTP methods. It does not implement business logic;
+it only presents fetched data and sends requests to the backend for actions.
+
+The backend runs in a Kubernetes cluster as a kubernetes service. Alternatively, it may run anywhere
+outside of the cluster, given that it can connect to the master. The backend is a HTTP server that
+proxies data requests to appropriate remote backends (e.g., k8s apiserver or heapster) or implements
```

Comment:
> use case for non-transitive auths? Is this something hypothetical, or solution that people actually use?

x509 client certificates are a primary example, and yes, people use them :)

to support non-transitive methods like this, you have a few options:
* the intermediate backend handles auth/authz, then escalates to the API using its own credentials
* the intermediate backend handles auth, then makes calls to the API using its own credentials, on behalf of the user (requires delegation support in the API)